src/main/java/org/broadinstitute/hellbender/tools/coveragemodel/linalg/FourierLinearOperator.java

```diff
@@ -0,0 +1,99 @@
+ */
+public abstract class FourierLinearOperator<V> extends GeneralLinearOperator<V> {
+
+    protected final int dimension;
+    protected final int fftSize;
+    protected final double[] fourierFactors;
+    protected final DoubleFFT_1D fftFactory;
+
+    protected final boolean promoteToPowerOfTwo;
+
+    /**
+     * Constructs the linear operator of a given {@code dimension} for given Fourier factors
+     * Note: the length of {@code fourierFactors} must be equal to {@code floor(dimension/2) + 1}
+     *
+     * @param dimension dimension of the linear space on which the linear operator acts
+     * @param fourierFactors Fourier factors of the linear operator
+     * @param promoteToPowerOfTwo promote the dimension to the power of two
+     * @throws IllegalArgumentException if {@code fourierFactors.length != dimension/2 + 1}
+     */
+    public FourierLinearOperator(final int dimension,
+                                 @Nonnull final double[] fourierFactors,
+                                 final boolean promoteToPowerOfTwo) {
+        ParamUtils.isPositive(dimension - 1, "The dimension of the linear operator " +
+                "must be >= 2.");
+        Utils.validateArg(fourierFactors.length == dimension/2 + 1, "The length of Fourier factors must be equal" +
+                " to floor(dimension/2) + 1");
+        if (!promoteToPowerOfTwo || isPowerOfTwo(dimension)) {
+            /* dimension is already power of 2, turn off the flag */
+            this.promoteToPowerOfTwo = false;
+            this.fftSize = dimension;
+            this.dimension = dimension;
+            this.fourierFactors = fourierFactors.clone();
+        } else {
+            this.promoteToPowerOfTwo = true;
+            this.fftSize = smallestPowerOfTwoGreaterThan(dimension);
+            this.dimension = dimension;
+            this.fourierFactors = nearestNeighborUniform1DInterpolate(fourierFactors, fftSize/2 + 1);
+        }
+        fftFactory = new DoubleFFT_1D(fftSize);
+    }
+
+    @Override
+    public int getRowDimension() { return dimension; }
+
+    @Override
+    public int getColumnDimension() { return dimension; }
+
+    @Override
+    public boolean isTransposable() {
+        return true;
+    }
+
+    public int getFFTSize() { return fftSize; }
+
+    @Override
+    public V operateTranspose(@Nonnull final V x)
+            throws DimensionMismatchException, UnsupportedOperationException {
+        return operate(x);
+    }
+
+    public double[] getFourierFactors() {
+        return fourierFactors.clone();
+    }
+
+    /**
+     * Generate Fourier factors for a midpass filter
+     *
+     * @param dimension dimension of the signal
+     * @param freqLowerCutoff lower frequency cutoff (excluded)
+     * @param freqUpperCutoff upper frequency cutoff (included)
+     * @return Fourier factors
+     */
+    public static double[] getMidpassFilterFourierFactors(final int dimension, final int freqLowerCutoff,
```

Comment:
I didn't want to create a Fourier utils class just for a single method. I will move it to `GATKProtectedMathUtils`, fine?